src/models/hooks/nationalCenter.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable import/prefer-default-export */
+const { Op } = require('sequelize');
+const { EVENT_REPORT_STATUSES } = require('@ttahub/common');
+
+const afterDestroy = async (sequelize, instance) => {
+  //   first, we need to see if there is a national center that maps to this one
+  const newNationalCenter = await sequelize.models.NationalCenter.findOne({
+    where: { mapsTo: instance.id },
+  });
+
+  const sessionReports = await sequelize.models.SessionReportPilot.findAll({
```

Comment:
An early out should be checked here to not even check for sessions if no mapped value is found.

Reply:
Updated the code with comments explaining what is going on